std/format.d

```diff
@@ -4362,14 +4362,44 @@ T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     return parse!T(input);
 }
 
+private T rawRead(T, Range)(ref Range input)
+    if (isSomeString!Range || ElementType!(Range).sizeof == 1)
+{
+    union X
+    {
+        ubyte[T.sizeof] raw;
+        T typed;
+    }
+    X x;
+    foreach (i; 0 .. T.sizeof)
+    {
+        static if (isSomeString!Range)
+        {
+            x.raw[i] = input[0];
+            input = input[1 .. $];
+        }
+        else
+        {
+            // TODO: recheck this
+            x.raw[i] = cast(ubyte) input.front;
```

Comment:
To avoid this cast and clarify the possibilities, you may put this constraint for the function:

`if (is(Unqual!(ElementType!Range) == char) || is(Unqual!(ElementType!Range) == byte) || is(Unqual!(ElementType!Range) == ubyte))`